lib/teiserver/coordinator/coordinator_lib.ex

```diff
@@ -2,166 +2,106 @@ defmodule Teiserver.Coordinator.CoordinatorLib do
+  any user set to ready is sent a ring from the coordinator", :host},
+      {"settag", ["key", "value"], "Sets a battletag of <key> to <value>", :host},
+      {"speclock", ["user"], "Locks the user into a spectator role. Can be reverted with the unban command.", :host},
+      {"forceplay", ["user"], "Forces the user into a player position.", :host},
+      {"rename", ["new name"],"Renames the lobby to the name given.", :host},
+      {"timeout", ["user", "reason, default: You have been given a timeout on the naughty step"], "Bans the user from the lobby for 1 game, will expire once the game is over.", :host},
+      {"lobbykick ", ["user"], "Kicks the user from the lobby.", :host},
+      {"lobbyban", ["user", "reason , default: None given"], "Bans the user from the lobby but not the server, will refresh on !rehost", :host},
+      {"lobbybanmult", ["user"], "Bans all users separated by spaces (from the lobby, not the game)
+  If you want to add a reason, add a `!!` to the end of the player list, anything after that will be the reason.", :host},
+      {"unban", ["user"],"Removes the user from the lobby banlist.", :host},
+      {"forcespec", ["user"], "Moves the user to spectators and bans them from becoming a player.", :host},
+
+      #---- :moderators only ----
+      {"success", [], "Sends a \"!y\" message from every player to the game host to make a vote pass.", :moderator},
+      {"playerlimit", ["limit"], "Sets a new player limit for this specific host.", :moderator},
+      {"check", ["user"], "Performs a smurf check against the user mentioned and sends you the result.", :moderator},
+      {"pull", ["user"], "Pulls a given user into the battle.", :moderator},
+      {"dosplit", [], "Completes the lobby split now.", :moderator},
+      {"cancelsplit", [], "Cancels the lobby split.", :moderator},
+      {"vip", ["name"], "Places that user at the front of the queue. This command will always output it's use even if used with the % operator.", :moderator},
+      {"reset", [], "Resets the coordinator bot for this lobby to the default.", :moderator},
+    ]
     # $command - Coordinator command
     # $%command - Don't echo command back to chat
+  result = commands
+    |> Enum.filter(fn {cmd, _args, _desc, group} ->
+      (cmd == command or command == "") and can_use?(user, host, group)
+    end)
 
-    [
-      everybody,
-      (if User.is_moderator?(user) or host, do: hosts, else: ""),
-      (if User.is_moderator?(user), do: moderators, else: "")
-    ]
-    |> Enum.join("\n")
+  case result do
+    [{cmd, args, desc, _group}] ->
+      arg_str = args |> Enum.map(fn a -> "<#{a}>" end) |> Enum.join(" ")
+      "#{cmd} #{arg_str}\n#{desc}"
+    _ ->
+      if command != "" do
+        "No commands matching that filter"
+      else
+
+
+        Enum.map(result, fn {cmd, args, desc, _group} ->
+        arg_str = args |> Enum.map(fn a -> "<#{a}>" end) |> Enum.join(" ")
+        "\n#{cmd} #{arg_str}\n#{desc}\n" end)
```

Comment:
the `end)` should be on it's own line as the opening is on a different line

Reply:
Ok